--- /usr/lib/python2.7/dist-packages/salt/grains/core.py	2017-06-06 00:17:57.315185933 -0500
+++ /usr/lib/python2.7/dist-packages/salt/grains/core.py.new	2017-06-06 00:15:12.837296917 -0500
@@ -1253,6 +1253,10 @@
                             )
                     elif salt.utils.which('supervisord') in init_cmdline:
                         grains['init'] = 'supervisord'
+                    elif init_bin.endswith('dumb-init'):
+                        grains['init'] = 'dumb-init'
+                    elif init_bin.endswith('tini'):
+                        grains['init'] = 'tini'
                     else:
                         log.info(
                             'Could not determine init system from command line: ({0})'
